src/pages/gallery.js

```diff
@@ -17,134 +17,101 @@ import FailedSearch from '../assets/images/failed-search.svg'
 import Dropdown from '../components/dropdown';
 import {faSearch} from "@fortawesome/free-solid-svg-icons/faSearch";
 import {useWindowWidth} from "@react-hook/window-size";
+import {OrderType, OrderDirection} from "../store/api";
+
+const SEARCH_STATUS = {
+  NoSearch: 'NoSearch',
+  Searching: 'Searching',
+  Success: 'Success',
+  Failed: 'Failed',
+}
 
 export default function Gallery() {
   const dispatch = useDispatch()
 
-  // Meanwhile get all the events
-  useEffect(() => {
-    dispatch(fetchIndexData());
-  }, []); /* eslint-disable-line react-hooks/exhaustive-deps */
-
-  const events  = useSelector(selectRecentEvents)
+  const events  = useSelector(selectEvents)
   const indexFetchStatus = useSelector(selectIndexFetchStatus)
+  const totalResultsAmount = useSelector(selectTotalResults)
+  const invalidEventsAmount = useSelector(state => state.events.currentInvalidResults)
 
   const [items, setItems] = useState(events)
-  const [search, setSearch] = useState(undefined);
-  const [length, setLength] = useState(40);
+  const [searchStatus, setSearchStatus] = useState(SEARCH_STATUS.NoSearch);
+  const [searchValue, setSearchValue] = useState('');
+  const [page, setPage] = useState(0);
+  const [moreToLoad, setMoreToLoad] = useState(true)
 
-  const orderTypeOptions = ['Date', 'Id', 'City', 'Holders', 'Transfers']
-  const [orderType, setOrderType] = useState(orderTypeOptions[0]);
-  const orderDirectionOptions = ['High to Low', 'Low to High']
-  const [orderDirection, setOrderDirection] = useState(orderDirectionOptions[0]);
+  const initialOrderType = OrderType.date
+  const initialOrderDirection = OrderDirection.descending
+  const [orderType, setOrderType] = useState(initialOrderType)
+  const [orderDirection, setOrderDirection] = useState(initialOrderDirection)
```

Comment:
@pablojp This Component needs to be refactored at some point. It's doing too many things and handling a lot of logic. We are also using too many `useStates` and this component is probable re-rendered a lot of times.